e2e-test/java/io/cdap/plugin/slt/stepsdesign/StepDefinitionSLT.java

```diff
@@ -0,0 +1,471 @@
+  static String color;
+  private static AssertionHelper assertionHelper = new AssertionHelper();
+  private static ElementHelper elementHelper = new ElementHelper();
+  private static WaitHelper waitHelper = new WaitHelper();
+  private static PluginPropertyUtils pluginPropertyUtils = new PluginPropertyUtils();
+
+
+  static {
+    sltLocators = SeleniumHelper.getPropertiesLocators(SLTLocators.class);
+    Properties prop = new Properties();
+    try {
+      prop.load(new FileInputStream("src/e2e-test/resources/Google_SAP_Connection.properties"));
+    } catch (IOException e) {
+      logger.error("Failed while reading SAP properties file" + e);
+    }
+    for (String property : prop.stringPropertyNames()) {
+      connection.put(property, prop.getProperty(property));
+    }
+  }
+
+  @Then("^Set Dummy MTID")
+  public void setDummyMTID() {
+    mtId = "02K";
+  }
+
+
+  @Given("^Open CDF replication and initiate pipeline creation")
+  public void open_cdf_replication() throws IOException, InterruptedException {
+    openCdf();
+    SeleniumDriver.getDriver().get("http://localhost:11011/cdap/ns/default/replication/create");
+    sltActions.enterName(UUID.randomUUID().toString().replaceAll("-", ""));
+    sltActions.clickNextButton();
+    isErrorPresent = false;
+    bqRecordCount = 0;
+    sapRecordsCount = 0;
+  }
+
+  @When("Select SAP SLT as source and fill connection parameters")
+  public void source_is_sap_slt() {
+    sltActions.clickSltPlugin();
+    sltActions.enterMandatoryParameters(
+      "gcpProjectId",
+      "gcsDataPath",
+      mtId,
+      "sapJcoLibGcsPath",
+      "jcoclienthost",
+      "jcoclient",
+      "jcoclientsysnr",
+      "jcoclientuser",
+      "jcoclientpasswd");
+  }
+
+  @When("Select SAP SLT as source")
+  public void source_is_sapslt() {
+    sltActions.clickSltPlugin();
+  }
+
+  @Then("Verify that after setting SLT parameter {string} as {string} plugin is throws error {string}")
+  public void userIsAbleToSetSLTParameterAsAndGettingRowForWrongInput(
+    String inputParameter, String inputValue, String errorMessage) {
+    errorExist = false;
+    elementHelper.replaceElementValue(sltLocators.inputParameter(inputParameter), inputValue);
+    sltActions.clickNextButton();
+    errorExist = pluginPropertyUtils.errorProp(errorMessage).
+      toLowerCase().contains(SLTLocators.rowError.getText().toLowerCase());
+    color = sltActions.getRowErrorBorderColor();
+    Assert.assertTrue(errorExist);
+    BeforeActions.scenario.write("Color of the text box" + color);
+    Assert.assertTrue(color.toLowerCase().contains("rgb(209, 86, 104)"));
+  }
+
+  @Then("Validate SLT parameter {string} can be set as {string}")
+  public void userIsAbleToSetSLTParameter(
+    String inputParameter, String inputValue) {
+    elementHelper.replaceElementValue(sltLocators.inputParameter(inputParameter), inputValue);
+  }
+
+  @Then("Validate Replicate Existing Data can be set to false")
+  public void replicate_existing_data() {
```

Comment:
->validateReplicateExistingDataState

Check all step definition's method names. Should align with step description